synapse/handlers/groups_local.py

```diff
@@ -37,9 +37,17 @@ def f(self, group_id, *args, **kwargs):
             )
         else:
             destination = get_domain_from_id(group_id)
-            return getattr(self.transport_client, func_name)(
+            d = getattr(self.transport_client, func_name)(
                 destination, group_id, *args, **kwargs
             )
+            def h(failure):
+                failure.trap(HttpResponseException)
+                e = failure.value
+                if e.code >= 400 and e.code < 500:
+                    raise SynapseError(e.code, e.msg)
```

Comment:
note that `HttpResponseException` has a `to_synapse_error` method which you probably want here.